src/JWT.ts

```diff
@@ -180,14 +203,29 @@ export async function createJWT(
       throw new Error('JWT expiresIn is not a number')
     }
   }
-  const signingInput: string = [
-    encodeSection(header),
-    encodeSection({ ...timestamps, ...payload, iss: issuer })
-  ].join('.')
+  const fullPayload = { ...timestamps, ...payload, iss: issuer }
+  return createJWS(fullPayload, signer, header)
+}
 
-  const jwtSigner: SignerAlgorithm = SignerAlgorithm(header.alg)
-  const signature: string = await jwtSigner(signingInput, signer)
-  return [signingInput, signature].join('.')
+/**
+ *  Verifies given JWS. If the JWS is valid, returns the public key that was used to sign the JWS.
+ *
+ *  @example
+ *  const pubkey = verifyJWT('eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJyZXF1Z....', { publicKeyHex: '0x12341...' })
+ *
+ *  @param    {String}                          jws         A JWS string to verify
+ *  @param    {Array<PublicKey> | PublicKey}    pubkeys     The public keys used to verify the JWS
+ *  @return   {PublicKey}                       The public key used to sign the JWS
+ */
+export function verifyJWS (jws: string, pubkeys: PublicKey | PublicKey[]) {
```

Comment:
The return type is missing, please mention `PublicKey`
```suggestion
export function verifyJWS (jws: string, pubkeys: PublicKey | PublicKey[]): PublicKey {
```

Also, this should be used in the `verifyJWT` method to not duplicate functionality.
I think it fits instead of the direct call to `VerifierAlgorithm` at L271
```
  const signer: PublicKey = VerifierAlgorithm(header.alg)(
    data,	    data,
    signature,	    signature,
    authenticators	    authenticators
  )	  )
```

Reply:
Yeah, I looked at doing this, but it would result in two calls to `decodeJWT`. I can change it if thats ok though.

Reply:
2 calls to decode is not a big problem in the long run; especially since decoding is fairly simple and does not involve any external calls.
The risk with code duplication is that if one side changes it becomes inconsistent.